crypto/x509/t_x509.c

```diff
@@ -200,9 +200,11 @@ int X509_print_ex(BIO *bp, X509 *x, unsigned long nmflags,
         }
     }
 
-    if (!(cflag & X509_FLAG_NO_EXTENSIONS))
-        X509V3_extensions_print(bp, "X509v3 extensions",
-                                X509_get0_extensions(x), cflag, 8);
+    if (!(cflag & X509_FLAG_NO_EXTENSIONS)
+        && !X509V3_extensions_print(bp, "X509v3 extensions",
+                                    X509_get0_extensions(x),
+                                    cflag & ~X509_FLAG_EXTENSIONS_ONLY_KID, 8))
```

Comment:
Why mask out `X509_FLAG_EXTENSIONS_ONLY_KID` here?

Reply:
I did this because I had the problem that for printing the first cert the output of the full list of extensions went wrong,
but I just found that the masking can be restricted to the more specialized `x509_print_ex_brief()` at the point when it calls  `X509_print_ex()`.